judgels-backends/judgels-commons/judgels-service-core/src/main/java/judgels/service/hibernate/Job.java

```diff
@@ -0,0 +1,33 @@
+package judgels.service.hibernate;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.context.internal.ManagedSessionContext;
+
+public abstract class Job implements Runnable {
```

Comment:
This should be inside `judgels-service-persistence`, which is the base module for a Judgels service with persistence.

(You don't need the extra hibernate dependency in build.gradle anymore)